channeldb/db.go

```diff
@@ -549,6 +555,14 @@ func (d *DB) FetchChannel(chanPoint wire.OutPoint) (*OpenChannel, error) {
 					return err
 				}
 
+				// Don't return the channel if the commitment tx
+				// has already confirmed.
+				if channel.HasChanStatus(
+					ChanStatusCommitConfirmed,
+				) {
+					return nil
+				}
+
```

Comment:
This is to keep `FetchChannel` backwards compatible. But maybe better to perform this check at the various call sites.